MapleServer2/Config/ConfigHandler.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace MapleServer2.Config
+{
+    public class ConfigHandler
+    {
+        private Dictionary<string, string> dictionary = new Dictionary<string, string>();
+
+        private static ConfigHandler instance = null;
+        public static ConfigHandler Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    instance = new ConfigHandler("config.ini");
+                }
+                return instance;
+            }
+        }
+
+        public string commentDelimiter { get; set; }
+        private string file;
+
+        public string this[string section, string key] => GetValue(section, key);
+
+        public ConfigHandler(string file, string commentDelimiter = ";")
+        {
+            this.commentDelimiter = commentDelimiter;
+            this.file = file;
+        }
+
+        public void Init(string value)
+        {
+            file = null;
+            dictionary.Clear();
+            if (File.Exists(value))
```

Comment:
Maybe do an early return if the file doesn't exist by inverting this if statement